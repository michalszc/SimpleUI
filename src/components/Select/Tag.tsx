import React, { FC, MouseEventHandler } from "react";
import styled from "styled-components";
import Styles, { StyleProps } from "../../utils/styles";
import { RxCross1 } from 'react-icons/rx';
import { Option } from ".";

export interface TagProps extends StyleProps {
    /**
     * Object of Option type {
     *  index: string
     *  value: string
     * }
     */
    value: Option;
    /**
     * onClick event handler -> fire when 'x' icon has been clicked
     */
    onClick: MouseEventHandler<HTMLAnchorElement>;
}

const StyledTag = styled.div`
    width: auto;
    height: 40%;
    background-color: #3ea8ff9a;
    padding: 8px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-radius: 5px;
    font-family: sans-serif;
    font-size: .9em;
    color: white;
    margin-right: 5px;
`;

const Cross = styled.a`
    cursor: pointer;
    padding-left: 5px;
`;

const Tag: FC<TagProps> = ({ value, onClick , ...props }) => {
    return(
        <Styles {...props}>
            <StyledTag key={value.index}>
                {value.value}
                <Cross onClick={onClick}><RxCross1 color="white" /></Cross>
            </StyledTag>
        </Styles>
    );
};

export default Tag;

