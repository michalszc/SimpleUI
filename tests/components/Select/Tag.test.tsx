import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Tag from "../../../src/components/Select/Tag";

describe("Tag", () => {
    const mockFn = jest.fn();
    
    test("renders Tag component", () => {
        const container = render(<Tag value="test"/>);
        expect(container).toMatchSnapshot();
    });

    test("should fire onclick event", () => {
        const { getByRole } = render(<Tag value="test" onClick={mockFn}/>);
        const cross = getByRole('link');
        fireEvent.click(cross);
        expect(mockFn).toBeCalled();
    });
});
