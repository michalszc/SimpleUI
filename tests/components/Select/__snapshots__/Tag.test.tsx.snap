// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders Tag component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe kHAQtY simpleui-tag"
      >
        test
        <a
          class="sc-gswNZR eXUciC"
          role="link"
        >
          <svg
            color="white"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: white;"
            viewBox="0 0 15 15"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe kHAQtY simpleui-tag"
    >
      test
      <a
        class="sc-gswNZR eXUciC"
        role="link"
      >
        <svg
          color="white"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 15 15"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
